src/ContainerRegistryHelper.ts

```diff
@@ -0,0 +1,56 @@
+import * as os from 'os';
+import { Utility } from './Utility';
+import { GitHubActionsToolHelper } from './GitHubActionsToolHelper';
+
+const toolHelper = new GitHubActionsToolHelper();
+const util = new Utility();
+
+export class ContainerRegistryHelper {
+    /**
+     * Authorizes Docker to make calls to the provided ACR instance using username and password.
+     * @param acrName - the name of the ACR instance to authenticate calls to
+     * @param acrUsername - the username for authentication
+     * @param acrPassword - the password for authentication
+     */
+    public async loginAcrWithUsernamePassword(acrName: string, acrUsername: string, acrPassword: string) {
+        toolHelper.writeDebug(`Attempting to log in to ACR instance "${acrName}" with username and password credentials`);
+        try {
+            let dockerTool = await toolHelper.which("docker", true);
+            await util.executeAndThrowIfError(`${dockerTool} login --username ${acrUsername} --password ${acrPassword} ${acrName}.azurecr.io`, [] , Buffer.from(acrPassword));
+        } catch (err) {
+            toolHelper.writeError(`Failed to log in to ACR instance "${acrName}" with username and password credentials`);
+            throw err;
+        }
+    }
+
+    /**
+     * Authorizes Docker to make calls to the provided ACR instance using an access token that is generated via
+     * the 'az acr login --expose-token' command.
+     * @param acrName - the name of the ACR instance to authenticate calls to.
+     */
+    public async loginAcrWithAccessTokenAsync(acrName: string) {
+        toolHelper.writeDebug(`Attempting to log in to ACR instance "${acrName}" with access token`);
+        try {
+            let commandLine = os.platform() === 'win32' ? 'pwsh' : 'bash';
+            await util.executeAndThrowIfError(`${commandLine} -c "CA_ADO_TASK_ACR_ACCESS_TOKEN=$(az acr login --name ${acrName} --output json --expose-token --only-show-errors | jq -r '.accessToken'); docker login ${acrName}.azurecr.io -u 00000000-0000-0000-0000-000000000000 -p $CA_ADO_TASK_ACR_ACCESS_TOKEN > /dev/null 2>&1"`);
```

Comment:
Any reason we need to define a `dockerTool` variable for the other two functions when calling the `docker` command, but for this command that we're setting up, we're just calling `docker` directly?

Reply:
We do not need `dockerTool` in a separate variable. We are able to call `docker` directly.